File: windows/src/AgentIsland/Model/QuotaAlarmStore.cs
using System.ComponentModel;
using AgentIsland.Core;

namespace AgentIsland.Model;

/// Whether hitting 100% on a 5-hour or weekly window raises the full-screen
/// "out of quota until &lt;time&gt;" alarm. Some people only want auto-resume and
/// treat the quota popup as low-priority — turning this off silences the
/// exhaustion alarm while leaving turn alarms untouched. Default ON, so the
/// pre-setting behavior is preserved for everyone who never opens Settings.
public sealed class QuotaAlarmStore : INotifyPropertyChanged
{
    private const string Key = "AgentIsland.quotaAlarmEnabled";

    public static QuotaAlarmStore Shared { get; } = new();

    private bool _enabled;

    public event PropertyChangedEventHandler? PropertyChanged;

    private QuotaAlarmStore()
    {
        // Missing key → ON. Preferences.Get<bool?> returns null when unset,
        // so the ?? keeps existing users who never touched the setting on the
        // prior behavior (a plain bool default would silence it for them).
        _enabled = Preferences.Get<bool?>(Key) ?? true;
    }

    public bool Enabled
    {
        get => _enabled;
        set
        {
            if (_enabled == value) return;
            _enabled = value;
            Preferences.Set(Key, value);
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(Enabled)));
        }
    }
}

File: Sources/Model/QuotaAlarmStore.swift
import Foundation

/// Whether hitting 100% on a 5-hour or weekly window raises the full-screen
/// "out of quota until <time>" alarm. Some people only want auto-resume and
/// treat the quota popup as low-priority — turning this off silences the
/// exhaustion alarm while leaving turn alarms untouched. Default ON, so the
/// pre-setting behavior is preserved for everyone who never opens Settings.
@MainActor
final class QuotaAlarmStore: ObservableObject {
    static let shared = QuotaAlarmStore()

    private static let key = "AgentIsland.quotaAlarmEnabled"

    @Published var enabled: Bool {
        didSet { UserDefaults.standard.set(enabled, forKey: Self.key) }
    }

    private init() {
        // Missing key → ON (bool(forKey:) would default false and silence it
        // for existing users who never touched the setting).
        if UserDefaults.standard.object(forKey: Self.key) == nil {
            enabled = true
        } else {
            enabled = UserDefaults.standard.bool(forKey: Self.key)
        }
    }
}
